index.rst

```diff
@@ -1,29 +1,481 @@
+     - single-block
+     - | Serialized :class:`~pipeline_graph.PipelineGraph`, including
+       | all configuration.
+   * - ``bipartite_edges``
+     - single-block
+     - Edge list for a dataset-quantum-dataset graph.
+   * - ``quanta``
+     - multi-block
+     - | UUIDs, task labels, data IDs, status, and exception information for
+       | each quantum; log and metadata dataset UUIDs and addresses (into
+       | the ``logs`` and ``metadata`` components).
+   * - ``datasets``
+     - multi-block
+     - | UUIDs, dataset type names, data IDs, and existence status for each
+       | each dataset (except log and metadata datasets).
+   * - ``logs``
+     - multi-block
+     - The content of all butler log datasets (one dataset per block).
+   * - ``metadata``
+     - multi-block
+     - The content of all butler metadata datasets (one dataset per block).
+   * - ``quantum_addresses``
+     - address (sorted)
+     - | UUID to integer ID mapping for all quanta; byte offsets and sizes
+       | for the ``quanta``, ``logs``, and ``metadata`` components.
+   * - ``dataset_addresses``
+     - address (sorted)
+     - | UUID to integer ID mapping for all datasets; byte offsets and sizes
+       | for the ``datasets`` component.
+
+Provenance graphs are designed to be ingested into the butler as a single datastore artifact that backs multiple datasets using a sophisticated :class:`~lsst.daf.butler.Formatter`:
+
+``run_provenance``
+  A thin wrapper around a :class:`networkx.MultiDiGraph` with quantum and dataset UUID keys, as a dataset type with no dimensions and hence a single instance for each :attr:`~lsst.daf.butler.CollectionType.RUN` collection.
+  This is primarily backed by the ``bipartite_edges``, ``quantum_addresses``, and ``dataset_addresses`` components, but an extensive suite of storage class parameters would be used to add extra information to some or all graph nodes (e.g. quantum status, dataset existence, even logs or metadata) from other components.
+  Higher-level provenance-query functionality would largely be implemented on top of parameter-modified reads of this dataset.
+
+``run_provenance.metadata``
+  A dictionary of basic information about the run, including submission host, begin and end timestamps (for the run as a whole), submitting user(s), etc.
+
+``run_provenance.pipeline_graph``
+  A :class:`~pipeline_graph.PipelineGraph` instance.
+  This includes the content of all per-task ``_config_`` datasets and should ultimately supersede them.
+  Making this a component dataset cuts down (slightly) on the number of butler dataset entries needed for provenance.
+
+``{task}_provenance``
+  A simple dataclass that combines quantum status information and optionally dataset existence information, logs, and metadata for a single quantum (as separate components), with a different dataset type for each task label (and the dimensions of that task).
+  The UUID of these datasets are always exactly the UUIDs of their quanta.
```

Comment:
Have to consider how this can be related to the `lsst.daf.butler.DatasetProvenance` class.

Reply:
I think it's cleanest if we just never put a quantum UUID into a `DatasetProvenance`, but this is one of the things that might get messy if we retire `_metadata` and `_log` datasets in favor of `<task>_provenance.(metadata|log)`.